control/diagnostics_tool/src/steering_module_state.c

```diff
@@ -0,0 +1,130 @@
+/************************************************************************/
+/* The MIT License (MIT) */
+/* ===================== */
+
+/* Copyright (c) 2016 PolySync Technologies, Inc.  All Rights Reserved. */
+
+/* Permission is hereby granted, free of charge, to any person */
+/* obtaining a copy of this software and associated documentation */
+/* files (the “Software”), to deal in the Software without */
+/* restriction, including without limitation the rights to use, */
+/* copy, modify, merge, publish, distribute, sublicense, and/or sell */
+/* copies of the Software, and to permit persons to whom the */
+/* Software is furnished to do so, subject to the following */
+/* conditions: */
+
+/* The above copyright notice and this permission notice shall be */
+/* included in all copies or substantial portions of the Software. */
+
+/* THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES */
+/* OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND */
+/* NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT */
+/* HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, */
+/* WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING */
+/* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR */
+/* OTHER DEALINGS IN THE SOFTWARE. */
+/************************************************************************/
+
+
+/**
+* @file steering_module_state.c
+* @brief Steering module Source.
+*
+*/
+
+
+
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "macros.h"
+#include "can_monitor.h"
+#include "system_state.h"
+#include "steering_module_state.h"
+
+
+
+
+// *****************************************************
+// static declarations
+// *****************************************************
+
+
+//
+static int analyze_command_frame(
+        steering_module_state_s * const state,
```

Comment:
Same comment as earlier: just pass in the buffer instead of the whole can frame.